packages/opencensus-core/src/trace/model/tracer-base.ts

```diff
@@ -0,0 +1,228 @@
+  private readonly IS_SAMPLED = 0x1;
+  /** A sampler used to make sample decisions */
+  sampler!: samplerTypes.Sampler;
+  /** A configuration for starting the tracer */
+  logger: loggerTypes.Logger = logger.logger();
+  /** A configuration object for trace parameters */
+  activeTraceParams: TraceParams;
+
+  /** Constructs a new TraceImpl instance. */
+  constructor() {
+    this.activeLocal = false;
+    this.activeTraceParams = {};
+  }
+
+  /** A propagation instance */
+  get propagation(): Propagation {
+    if (this.config && this.config.propagation) {
+      return this.config.propagation;
+    }
+    return noopPropagation;
+  }
+
+  /**
+   * Starts a tracer.
+   * @param config A tracer configuration object to start a tracer.
+   */
+  start(config: configTypes.TracerConfig): this {
+    this.activeLocal = true;
+    this.config = config;
+    this.logger = this.config.logger || logger.logger();
+    this.sampler =
+        SamplerBuilder.getSampler(config.samplingRate || DEFAULT_SAMPLING_RATE);
+    if (config.traceParams) {
+      this.activeTraceParams.numberOfAnnontationEventsPerSpan =
+          TraceParamsBuilder.getNumberOfAnnotationEventsPerSpan(
+              config.traceParams);
+      this.activeTraceParams.numberOfAttributesPerSpan =
+          TraceParamsBuilder.getNumberOfAttributesPerSpan(config.traceParams);
+      this.activeTraceParams.numberOfMessageEventsPerSpan =
+          TraceParamsBuilder.getNumberOfMessageEventsPerSpan(
+              config.traceParams);
+      this.activeTraceParams.numberOfLinksPerSpan =
+          TraceParamsBuilder.getNumberOfLinksPerSpan(config.traceParams);
+    }
+    return this;
+  }
+
+  /** Stops the tracer. */
+  stop(): this {
+    this.activeLocal = false;
+    return this;
+  }
+
+  /** Gets the list of event listeners. */
+  get eventListeners(): types.SpanEventListener[] {
+    return this.eventListenersLocal;
+  }
+
+  /** Indicates if the tracer is active or not. */
+  get active(): boolean {
+    return this.activeLocal;
+  }
+
+  /**
+   * Starts a root span.
+   * @param options A TraceOptions object to start a root span.
+   * @param fn A callback function to run after starting a root span.
+   */
+  startRootSpan<T>(options: types.TraceOptions, fn: (root: types.Span) => T):
+      T {
+    const spanContext: types.SpanContext = options.spanContext ||
+        {spanId: '', traceId: uuid.v4().split('-').join('')};
+    const parentSpanId = spanContext.spanId;
+    const traceId = spanContext.traceId;
+    const name = options.name || 'span';
+    const kind = options.kind || types.SpanKind.UNSPECIFIED;
+    const traceState = spanContext.traceState;
+
+    if (this.active) {
+      const sampleDecision = this.makeSamplingDecision(options, traceId);
+      if (sampleDecision) {
+        const rootSpan =
+            new RootSpan(this, name, kind, traceId, parentSpanId, traceState);
+        rootSpan.start();
+        return fn(rootSpan);
+      }
+    } else {
```

Comment:
Can we remove this else, and update the debug msg: `Tracer is inactive, starting new no record root span`?